### PR TITLE
Recompose and Refactor

### PR DESCRIPTION
**Pull request by kabirbaidhya**
* Add Recompose
* Minor changes on the filenames for consistency
* Refactor `StatusPanel` using hoc `withStatusInfo`
* Rename `Dashboard` to `StatusPage` as it's we're likely going to need a separate component for Dashboard which is shown after login later on.

### CODE REVIEW

File: src/components/dashboard/StatusPanel.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import * as websocket from '../../services/websocket';
+import * as statusService from '../../services/status';
+
+import Panel from '../commons/Panel';
+import ServiceList from './ServiceList';
+import Spinner from '../commons/Spinner';
+
+class StatusPanel extends Component {
+  componentDidMount() {
```
Comment (kabirbaidhya) [inline src/components/dashboard/StatusPanel.js]: @mesaugat @pratishshr @bishalshr Should we move all these methods along with the lifecycle ones to the HOC, converting this class to a function? What do you think?

File: src/components/hoc/status.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { update } from 'ramda';
+import { compose, withState, withHandlers } from 'recompose';
```
Comment (romitamgai) [inline src/components/hoc/status.js]: modules could be imported individually 
`import compose from 'recompose/compose'` or use babel-recompose-plugin

Comment (kabirbaidhya) [inline src/components/hoc/status.js]: But we have webpack 2 tree shaking too, that would do the same thing, only bundle the imported stuff.

File: src/components/dashboard/StatusPanel.js (hunk 1 above)
Comment (bishalshr) [inline src/components/dashboard/StatusPanel.js]: I would prefer not to use withState as I don't like the way of how setting state works. And its just my opinion. In the other side, it makes code cleaner and its the current trend. So, I think we can use it.

**Comment by pratishshr** [comment]
@kabirbaidhya Removed recompose to use plain HOCs, please verify.

**Pull request merged**